lib/Zonemaster/Backend/Validator.pm

```diff
@@ -211,6 +210,102 @@ sub jsonrpc_method {
     return joi->string->regex( $JSONRPC_METHOD_RE );
 }
 
+sub formats {
+    my ( $rpcapi ) = @_;
+    return {
+        domain => sub { check_domain( $rpcapi, @_ ) },
+        language_tag => sub { check_language_tag( $rpcapi, @_ ) },
+        ip => sub { check_ip( $rpcapi, @_ ) },
+        profile => sub { check_profile( $rpcapi, @_ ) },
+    };
+}
+
+sub check_domain {
+    my ( $rpcapi, $domain ) = @_;
```

Comment:
Have you considered making the `check_*()` methods only take the values of the individual configuration properties that they actually use? That way when you read the calling code its clearer what parts of the configuration are actually used for input and what parts are ignored.

Reply:
I have changed it to what you describe.